private/runtime/binding-spec.rkt

```diff
@@ -325,6 +353,56 @@
       (call-reconstruct-function (exp-state-pvar-vals st^) reconstruct-f)
       (exp-state-nest-state st^))]))
 
+; exp-state? (listof symbol?) -> (listof exp-state?)
+(define (exp-state-split/ellipsis st pvars)
+  (match st
+    [(exp-state pvar-vals nest-state)
+     (for/list ([env (env-split/ellipsis pvar-vals pvars)])
+       (exp-state env nest-state))]))
+
+; (hash symbol? (treeof syntax?)) (listof symbol?) -> (listof (hash symbol? (treeof syntax?)))
```

Comment:
It might be nice to have a "why" comment explaining the general strategy for expanding ellipses---this comment covers the "what" only. Maybe such a comment could sit at the top of all the helpers used?